src/transformers/models/idefics2/modeling_idefics2.py

```diff
@@ -923,10 +1000,15 @@ def __init__(self, config, layer_idx: int):
         self.n_latents = config.perceiver_config.resampler_n_latents
         self.depth = config.perceiver_config.resampler_depth
         self.rms_norm_eps = config.text_config.rms_norm_eps
+        attn_implementation = (
+            config._attn_implementation["perceiver_config"]
+            if config._attn_implementation["perceiver_config"] is not None
+            else "eager"
+        )
```

Comment:
Couldn't make Perceiever a PreTrained model which mean we can't call "dispatch_attn" when loading the perceiver module. A bit hacky, but since we know perceiver supports all attn types we can just dispatch to whatever was asked by the user. No need for warnings/errors, because hardware related errors will be raise when dispatching the vision/;m backbones

The main reason why perceiver couldn't be a pretrained model, is because it accepts a general config, not PerceiverConfig as input. Later it uses perceiver config in attention modules